externals/kyuubi-jdbc-engine/src/main/scala/org/apache/kyuubi/engine/jdbc/mysql/MySQLConnectionProvider.scala

```diff
@@ -0,0 +1,22 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kyuubi.engine.jdbc.mysql
+
+class MySQLConnectionProvider extends Mysql8ConnectionProvider {
```

Comment:
Again why not use `Mysql8ConnectionProvider` for connection provider directly? What's the key point for introducing such a dedicated provider?

Reply:
Or how about create an abstract MySQLConnectionProvider and make Mysql8ConnectionProvider as a subclass of it.

Reply:
> Again why not use `Mysql8ConnectionProvider` for connection provider directly? What's the key point for introducing such a dedicated provider?

I think the connection provider is used for creating JDBC connections. If different databases utilize the same connection provider, it might cause confusion for users. Using distinct connection providers would also facilitate future customizations of connections to Doris and MySQL.

Reply:
There are many MySQL protocol-compatible databases. We don't need separate connection providers for each of them, especially like  Doris/Starrocks which uses Mysql driver solely and does not even have its own JDBC driver.